### PR TITLE
Spells/Druid: Implement Incarnation tree of life talent

### LINKED ISSUE #29142: Druid/Spell: Incarnation: Tree of Life
### Description

https://www.wowhead.com/spell=33891/incarnation-tree-of-life

The problem is that the tree form must last 30 seconds. In TC, it is valid until canceled by the player, like the appearance of a cat, bear, etc.;

### Expected behaviour

Shapeshift into the Tree of Life, increasing healing done by 15%, increasing armor by 120%, and granting protection from Polymorph effects. Functionality of Rejuvenation, Wild Growth, Regrowth, and Entangling Roots is enhanced.

Lasts 30 sec. You may shapeshift in and out of this form for its duration.

### Steps to reproduce the problem

Just create druid and use this spell.

### Branch

master

### TC rev. hash/commit

https://github.com/TrinityCore/TrinityCore/commit/9ad227f0756aa91f9b772ecf13b40c1f38863e68

### Operating system

Debian 11

### Custom changes

None

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Implement incarnation tree of life talent

**Issues addressed:**

Closes #29142

**Tests performed:**

Tested in-game

### CODE REVIEW

**Comment by Nyr97** [comment]
More like a partial implementation of the talent, only casting for the appropriate setting of the aura is handled atm.

**Comment by aquadeus** [comment]
bonus parts are already applied

**Comment by Nyr97** [comment]
Does that mean that this PR already handles:

Wild Growth max. target is increased by 2.
Regrowth becomes an instant cast.
Rejuvenation's and Germination's (Talent) healing is increased by 50% and their mana cost is reduced by 30%.
Entangling Roots becomes an instant cast.

**Comment by aquadeus** [comment]
Wild growth extra targets was missing ty

File: src/server/scripts/Spells/spell_druid.cpp [hunk 1]
```diff
@@ -1648,16 +1649,18 @@ class spell_dru_wild_growth : public SpellScript
...
-        if (!ValidateSpellEffect({ { spellInfo->Id, EFFECT_2 } }) || spellInfo->GetEffect(EFFECT_2).IsEffect() || spellInfo->GetEffect(EFFECT_2).CalcValue() <= 0)
-            return false;
-        return true;
+        return ValidateSpellEffect({ { spellInfo->Id, EFFECT_1 }, { SPELL_DRUID_TREE_OF_LIFE, EFFECT_2 } })
+            && ValidateSpellInfo({ SPELL_DRUID_TREE_OF_LIFE });
     }
 
     void FilterTargets(std::list<WorldObject*>& targets)
     {
         targets.remove_if(RaidCheck(GetCaster()));
```
Comment (Nyr97) [inline src/server/scripts/Spells/spell_druid.cpp:1658]: This is wrong up to this day. Wild Growth (like many other AoE healing spells) do not remove targets that are outside the caster's group or raid group, they only prefer targets within.

Comment (aquadeus) [reply src/server/scripts/Spells/spell_druid.cpp:1658]: that spell is outdated (effect 2 doesnt exist) but that should be another pr

File: src/server/scripts/Spells/spell_druid.cpp [hunk 2]
```diff
@@ -734,6 +737,56 @@ class spell_dru_innervate : public SpellScript
...
+        return ValidateSpellInfo
+        ({
+            SPELL_DRUID_INCARNATION_TREE_OF_LIFE
+        });
+    }
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTarget()->HasAura(SPELL_DRUID_INCARNATION_TREE_OF_LIFE))
+            GetTarget()->RemoveAurasDueToSpell(SPELL_DRUID_INCARNATION_TREE_OF_LIFE);
```
Comment (Seyden) [inline src/server/scripts/Spells/spell_druid.cpp:756]: Just GetTarget()->RemoveAura(SPELL_DRUID_INCARNATION_TREE_OF_LIFE) is enough, no need to check with HasAura

File: src/server/scripts/Spells/spell_druid.cpp (hunk 1 above)
Comment (Shauren) [reply src/server/scripts/Spells/spell_druid.cpp:1612]: This PR is focused on Incarnation tree of life, not anything else, its fine to leave this as it is now

**Pull request merged**